web-bundle/src/main/java/com/graphhopper/resources/MapboxResource.java

```diff
@@ -0,0 +1,185 @@
+/*
+ *  Licensed to GraphHopper GmbH under one or more contributor
+ *  license agreements. See the NOTICE file distributed with this work for
+ *  additional information regarding copyright ownership.
+ *
+ *  GraphHopper GmbH licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except in
+ *  compliance with the License. You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.graphhopper.resources;
+
+import com.graphhopper.GHRequest;
+import com.graphhopper.GHResponse;
+import com.graphhopper.GraphHopperAPI;
+import com.graphhopper.MultiException;
+import com.graphhopper.util.StopWatch;
+import com.graphhopper.util.shapes.GHPoint;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.*;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.core.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import static com.graphhopper.util.Parameters.Routing.*;
+
+/**
+ * Provides an endpoint that is compatible with the Mapbox API v5
+ *
+ * See: https://www.mapbox.com/api-documentation/#directions
+ *
+ * @author Robin Boldt
+ */
+@Path("mapbox/directions/v5/mapbox")
+public class MapboxResource {
```

Comment:
Is it possible to use OSRM for the Mapbox SDK? If yes, can we rename to OSRMResource? If no, this is perfect already :)

Reply:
Actually I don't think that this is reasonably possible. The problem is that the OSRM endpoint does not generate voice instructions nor banner instructions which are both important for the MapboxSDK.